src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -131,81 +144,102 @@ public void start(Map<String, String> properties) {
+            new TimestampIncrementingTableQuerier(
+                dialect,
+                queryMode,
+                tableOrQuery,
+                topicPrefix,
+                null,
+                incrementingColumn,
+                offset,
+                timestampDelayInterval
+            )
+        );
       } else if (mode.equals(JdbcSourceTaskConfig.MODE_TIMESTAMP)) {
-        tableQueue.add(new TimestampIncrementingTableQuerier(
-            queryMode, tableOrQuery, topicPrefix, timestampColumn, null, offset,
-                timestampDelayInterval, schemaPattern, mapNumerics));
+        tableQueue.add(
+            new TimestampIncrementingTableQuerier(
+                dialect,
+                queryMode,
+                tableOrQuery,
+                topicPrefix,
+                timestampColumns,
+                null,
+                offset,
+                timestampDelayInterval
+            )
+        );
       } else if (mode.endsWith(JdbcSourceTaskConfig.MODE_TIMESTAMP_INCREMENTING)) {
-        tableQueue.add(new TimestampIncrementingTableQuerier(
-            queryMode, tableOrQuery, topicPrefix, timestampColumn, incrementingColumn,
-                offset, timestampDelayInterval, schemaPattern, mapNumerics));
+        tableQueue.add(
+            new TimestampIncrementingTableQuerier(
+                dialect,
+                queryMode,
+                tableOrQuery,
+                topicPrefix,
+                timestampColumns,
+                incrementingColumn,
+                offset,
+                timestampDelayInterval
+            )
+        );
       }
     }
 
-    stop = new AtomicBoolean(false);
-  }
-
-  private void createConnectionProvider() {
-    final String dbUrl = config.getString(JdbcSourceConnectorConfig.CONNECTION_URL_CONFIG);
-    final String dbUser = config.getString(JdbcSourceConnectorConfig.CONNECTION_USER_CONFIG);
-    final Password dbPassword
-        = config.getPassword(JdbcSourceConnectorConfig.CONNECTION_PASSWORD_CONFIG);
-    final int maxConnectionAttempts
-        = config.getInt(JdbcSourceConnectorConfig.CONNECTION_ATTEMPTS_CONFIG);
-    final long connectionRetryBackoff
-        = config.getLong(JdbcSourceConnectorConfig.CONNECTION_BACKOFF_CONFIG);
-    cachedConnectionProvider = new CachedConnectionProvider(
-        dbUrl,
-        dbUser,
-        dbPassword == null ? null : dbPassword.value(),
-        maxConnectionAttempts,
-        connectionRetryBackoff
-    );
+    running.set(true);
   }
 
   @Override
   public void stop() throws ConnectException {
-    if (stop != null) {
-      stop.set(true);
-    }
-    if (cachedConnectionProvider != null) {
-      cachedConnectionProvider.closeQuietly();
+    try {
+      running.set(false);
+      if (cachedConnectionProvider != null) {
+        cachedConnectionProvider.close();
+      }
+    } finally {
+      cachedConnectionProvider = null;
+      try {
+        if (dialect != null) {
+          dialect.close();
```

Comment:
Should we use `org.apache.kafka.common.utils.Utils.closeQuietly` ?

Reply:
I actually like the warning appearing on the same logger, so I just caught Throwable here and logged a warning.